src/analysis/retail/shaman/shared/Abilities.tsx

```diff
@@ -0,0 +1,384 @@
+        category: SPELL_CATEGORY.DEFENSIVE,
+        gcd: {
+          base: 1000,
+        },
+        cooldown: 180,
+      },
+      {
+        spell: SPELLS.FLAME_SHOCK.id,
+        category: SPELL_CATEGORY.ROTATIONAL,
+        cooldown: combatant.specId === SPECS.ENHANCEMENT_SHAMAN.id ? (haste) => 6 / (1 + haste) : 6,
+        gcd: {
+          base: 1500,
+        },
+        range: 40,
+      },
+      {
+        spell: SPELLS.GHOST_WOLF.id,
+        category: SPELL_CATEGORY.UTILITY,
+        gcd: {
+          base: 1500,
+        },
+        castEfficiency: {
+          suggestion: false,
+        },
+      },
+      {
+        spell: SPELLS.HEALING_SURGE.id,
+        gcd: {
+          base: 1500,
+        },
+        category: SPELL_CATEGORY.OTHERS,
+      },
+      {
+        spell: TALENTS.ASTRAL_SHIFT_TALENT.id,
+        enabled: combatant.hasTalent(TALENTS.ASTRAL_SHIFT_TALENT),
+        cooldown: 120 - combatant.getTalentRank(TALENTS.PLANES_TRAVELER_TALENT) * 30,
+        category: SPELL_CATEGORY.DEFENSIVE,
+      },
+      {
+        spell: TALENTS.CHAIN_LIGHTNING_TALENT.id,
+        enabled: combatant.hasTalent(TALENTS.CHAIN_LIGHTNING_TALENT),
+        category: SPELL_CATEGORY.ROTATIONAL_AOE,
+        gcd: {
+          base: 1500,
+        },
+      },
+      {
+        spell: TALENTS.FROST_SHOCK_TALENT.id,
+        enabled: true, // combatant.hasTalent(TALENTS.FROST_SHOCK_TALENT),
+        category: SPELL_CATEGORY.ROTATIONAL,
+        cooldown: (haste) =>
+          combatant.specId === SPECS.ENHANCEMENT_SHAMAN.id ? 6 / (1 + haste) : 0,
+        gcd: {
+          base: 1500,
+        },
+      },
+      {
+        spell: SPELLS.LIGHTNING_BOLT.id,
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: {
+          base: 1500,
+        },
+      },
+      {
+        spell: TALENTS.LAVA_BURST_TALENT.id,
+        charges: (c) => {
+          switch (c.specId) {
+            case SPECS.ELEMENTAL_SHAMAN.id:
+              return 1 + (c.hasTalent(TALENTS.ECHO_OF_THE_ELEMENTS_TALENT) ? 1 : 0);
+            case SPECS.ENHANCEMENT_SHAMAN.id:
+              return c.hasTalent(TALENTS.ELEMENTAL_BLAST_ENHANCEMENT_TALENT)
+                ? 0
+                : c.hasTalent(TALENTS.LAVA_BURST_TALENT)
+                  ? 1
+                  : 0;
+            default:
+              return 1;
+          }
+        },
+        enabled: (c) => {
+          switch (c.specId) {
+            case SPECS.ENHANCEMENT_SHAMAN.id:
+              return (
+                c.hasTalent(TALENTS.LAVA_BURST_TALENT) &&
+                !c.hasTalent(TALENTS.ELEMENTAL_BLAST_ENHANCEMENT_TALENT)
+              );
+            default:
+              return c.hasTalent(TALENTS.LAVA_BURST_TALENT);
+          }
+        },
+        cooldown: (_) => (combatant.hasBuff(TALENTS.ASCENDANCE_ELEMENTAL_TALENT.id) ? 0 : 8),
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: {
+          base: 1500,
+        },
+      },
```

Comment:
rather than having spec-specific logic for this spell here, this should just be added to the individual specs. it is a little bit of duplication, but makes it easier to keep track of. it also means that we don't need to add extra complexity to the core `Ability` type.

if you really want to keep this here, I will note that the spec and selected combatant are accessible in the shared analyzer so you can just use `combatant` here instead of needing to defer computation with the `(c) => { ... }` stuff.

Reply:
> rather than having spec-specific logic for this spell here, this should just be added to the individual specs. it is a little bit of duplication, but makes it easier to keep track of. it also means that we don't need to add extra complexity to the core Ability type.

My personal opinion is it's easier to maintain _overall_ if shared abilities are in one spot, even if the spec specific logic adds some complexity.

> if you really want to keep this here, I will note that the spec and selected combatant are accessible in the shared analyzer so you can just use combatant here instead of needing to defer computation with the (c) => { ... } stuff

Good idea, done. Also reverted the change to `Ability.ts` `enabled` as a result.